src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -261,6 +265,30 @@ public ExamScoresDTO calculateExamScores(Long examId) {
                         studentResult.overallPointsAchieved += achievedPoints;
                     }
 
+                    // collect points of first correction, if a second correction exists
+                    if (exam.getNumberOfCorrectionRoundsInExam() == 2 && !exercise.getIncludedInOverallScore().equals(IncludedInOverallScore.NOT_INCLUDED)) {
+                        Optional<Submission> latestSubmission = studentParticipation.findLatestSubmission();
+                        if (latestSubmission.isPresent()) {
+                            Submission submission = latestSubmission.get();
+                            // Check if second correction already started
+                            if (submission.getManualResults().size() > 1) {
+                                if (!scores.hasSecondCorrectionAndStarted) {
+                                    scores.hasSecondCorrectionAndStarted = true;
+                                }
+                                Result firstManualResult = submission.getFirstManualResult();
+                                double achievedPointsInFirstCorrection;
+                                if (firstManualResult != null) {
+                                    Double resultScore = firstManualResult.getScore();
+                                    achievedPointsInFirstCorrection = resultScore != null ? round(resultScore / 100.0 * exercise.getMaxPoints()) : 0.0;
```

Comment:
Is it possible for exercises to have a maxPoints of 0? That could potentially be a problem then

Reply:
You have to insert a number for maxPoints when creating a new exercise. So this way there should always be maxPoints. And since I do it the same way as for the second correction overallPoints some lines above, this should work